src/badguy/root.hpp

```diff
@@ -40,21 +41,26 @@ class Root final : public BadGuy
 
   virtual bool is_flammable() const override { return false; }
   virtual bool is_freezable() const override { return false; }
-  virtual bool is_snipable () const override { return false; }
+  virtual bool is_snipable() const override { return false; }
 
   virtual std::vector<Direction> get_allowed_directions() const override;
 
 private:
-  enum State { STATE_HATCHING, STATE_APPEARING, STATE_RETREATING };
-
   void construct(float delay = -1, bool play_sound = true);
 
+  enum State { STATE_HATCHING, STATE_APPEARING, STATE_RETREATING };
```

Comment:
I thought types came first

Reply:
Functions first, then enumerators and variables, so everything is fully organized.